### PR TITLE
Update RELEASES.md for 1.42.0

### PR DESCRIPTION
**Pull request by XAMPPRocky**
### [Rendered](https://github.com/XAMPPRocky/rust/blob/relnotes-1.42.0/RELEASES.md)

r? @Mark-Simulacrum 
cc @rust-lang/release

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1,3 +1,80 @@
...
+  ```rust
+  fn foo(words: &[&str]) {
+      match words {
+          ["Hello", "World", "!", ..] => println!("Hello World!"),
+          ["Foo", "Bar", ..] => println!("Baz"),
+          rest => println!("{}", rest),
+      }
+  }
+  ```
+  <!-- TODO -->
```
Comment (jonas-schievink) [inline RELEASES.md]: What's this referring to?

File: RELEASES.md [hunk 2]
```diff
@@ -1,3 +1,80 @@
...
+- [Merge `TraitItem` & `ImplItem into `AssocItem`][67131]
```
Comment (jonas-schievink) [inline RELEASES.md]:
```suggestion
- [Merge `TraitItem` & `ImplItem` into `AssocItem`][67131]
```

File: RELEASES.md [hunk 3]
```diff
@@ -1,3 +1,80 @@
...
+
+Compiler
+--------
+- [Added tier 3\* support for `armv7a-none-eabi`.][68253]
```
Comment (jonas-schievink) [inline RELEASES.md]:
```suggestion
- [Added tier 2\* support for `armv7a-none-eabi`.][68253]
```

File: RELEASES.md [hunk 4]
```diff
@@ -1,3 +1,80 @@
...
+- [Added tier 3 support for `riscv64gc-unknown-linux-gnu`.][68037]
```
Comment (jonas-schievink) [inline RELEASES.md]: This was also made a tier 2 target in https://github.com/rust-lang/rust/pull/68339

File: RELEASES.md [hunk 5]
```diff
@@ -1,3 +1,80 @@
...
+
+Stabilized APIs
+---------------
+- [`DebugMap::key`]
+- [`DebugMap::value`]
+- [`ManuallyDrop::take`]
+- [`matches!`]
+- [`ptr::slice_from_raw_parts_mut`]
+- [`ptr::slice_from_raw_parts`]
+- [`wait_timeout_while`]
```
Comment (jonas-schievink) [inline RELEASES.md]: Link to https://doc.rust-lang.org/std/sync/struct.Condvar.html#method.wait_timeout_while is missing

```suggestion
- [`Condvar::wait_timeout_while`]
```

File: RELEASES.md [hunk 6]
```diff
@@ -1,3 +1,80 @@
...
+- [`wait_while`]
```
Comment (jonas-schievink) [inline RELEASES.md]:
```suggestion
- [`Condvar::wait_while`]
```

File: RELEASES.md [hunk 7]
```diff
@@ -1,3 +1,80 @@
...
+[66899]: https://github.com/rust-lang/rust/pull/66899/
+[66919]: https://github.com/rust-lang/rust/pull/66919/
+[66254]: https://github.com/rust-lang/rust/pull/66254/
+[`DebugMap::key`]: https://doc.rust-lang.org/stable/std/fmt/struct.DebugMap.html#method.key
+[`DebugMap::value`]: https://doc.rust-lang.org/stable/std/fmt/struct.DebugMap.html#method.value
+[`ManuallyDrop::take`]: https://doc.rust-lang.org/stable/std/mem/struct.ManuallyDrop.html#method.take
+[`matches!`]: https://doc.rust-lang.org/stable/std/macro.matches.html
+[`ptr::slice_from_raw_parts_mut`]: https://doc.rust-lang.org/stable/std/ptr/fn.slice_from_raw_parts_mut.html
+[`ptr::slice_from_raw_parts`]: https://doc.rust-lang.org/stable/std/ptr/fn.slice_from_raw_parts.html
+[`wait_while`]: https://doc.rust-lang.org/stable/std/sync/struct.Condvar.html#method.wait_while
```
Comment (jonas-schievink) [inline RELEASES.md]:
```suggestion
[`Condvar::wait_while`]: https://doc.rust-lang.org/stable/std/sync/struct.Condvar.html#method.wait_while
```

**Comment by ehuss** [comment]
I'm curious if people want to include https://github.com/rust-lang/cargo/pull/7700?  It is no longer necessary to specify `extern crate proc_macro;` in a proc-macro crate with Cargo.

**Comment by XAMPPRocky** [comment]
@ehuss Included.

File: RELEASES.md (hunk 2 above)
Comment (Centril) [reply RELEASES.md]:
```suggestion
- [Some *syntax-only* changes include:][67131]
   - `default` is syntactically allowed before items in `trait` definitions.
   - Items in `impl`s (i.e. `const`s, `type`s, and `fn`s) may syntactically leave out their bodies in favor of `;`.
   - Bounds on associated types in `impl`s are now syntactically allowed (e.g. `type Foo: Ord;`).
   - `...` (the C-variadic type) may occur syntactically directly as the type of any function parameter.
  
  These are still rejected *semantically*, but can be seen and parsed by procedural macros and conditional compilation.
```

File: RELEASES.md [hunk 8]
```diff
@@ -1,3 +1,86 @@
...
+- [You can now use `#[repr(transparent)]` on univariant `enum`s.][68122] Meaning
+  that you can create an enum that has the exact layout and ABI of the type
+  it contains.
+- [You can now use the slice pattern syntax with subslices.][67712] e.g.
+  ```rust
+  fn foo(words: &[&str]) {
+      match words {
+          ["Hello", "World", "!", ..] => println!("Hello World!"),
+          ["Foo", "Bar", ..] => println!("Baz"),
+          rest => println!("{}", rest),
```
Comment (CryZe) [inline RELEASES.md]: rest is a `&[&str]` which doesn't impl Display.

File: RELEASES.md [hunk 9]
```diff
@@ -1,3 +1,100 @@
+Version 1.42.0 (2020-03-12)
+==========================
+
+Language
+--------
+- [You can now use the slice pattern syntax with subslices.][67712] e.g.
```
Comment (Mark-Simulacrum) [inline RELEASES.md]: Should this just be "with slices"? I'm not sure subslices are a thing?

**Review by Mark-Simulacrum — APPROVED** [review]
Looking good to me.

File: RELEASES.md (hunk 9 above)
Comment (cuviper) [reply RELEASES.md]: See #67712 -- I think "with slices" doesn't convey what's new, which is using `..` to match a subset of the slice.

Comment (Mark-Simulacrum) [reply RELEASES.md]: Ah. Interesting. Okay, then this seems fine.

Comment (Centril) [reply RELEASES.md]: Subslices are a thing . :)

**Comment by Mark-Simulacrum** [comment]
I'm going to go ahead and approve this, I think we can get it in a bit early this cycle. And we can always have another PR if necessary :)

@bors r+ rollup=always

**Pull request merged**